diff --git a/src/main.sh b/src/main.sh
index 3f0a9e0..775021d 100755
--- a/src/main.sh
+++ b/src/main.sh
@@ -1393,9 +1393,15 @@ else
 			# Install the missing deps.
 			msg "$(gettext "Installing missing dependencies...")"
 
+			# Try using apt-get satisfy (it is not available in Debian 10 and lower)
 			if ! sudo "${SUDOARGS[@]}" -- apt-get satisfy "${APTARGS[@]}" -- "${predepends[@]}" "${depends[@]}" "${makedepends[@]}" "${checkdepends[@]}"; then
-				error "$(gettext "Failed to install missing dependencies.")"
-				exit "${E_INSTALL_DEPS_FAILED}"
+				warning "$(gettext "apt-get satisfy is not available, falling back to apt-get install.")"
+
+				# Fallback to apt-get install
+				if ! DEBIAN_FRONTEND=noninteractive sudo "${SUDOARGS[@]}" -- apt-get install -y "${APTARGS[@]}" -- "${predepends[@]}" "${depends[@]}" "${makedepends[@]}" "${checkdepends[@]}"; then
+					error "$(gettext "Failed to install missing dependencies using both satisfy and install.")"
+					exit "${E_INSTALL_DEPS_FAILED}"
+				fi
 			fi
 
 			# Get the list of packages that were just installed.